app/controllers/books_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  before_action :save_my_previous_url
+
+  def index
+    @book_count = Book.count
+    @categories = Category.select(:id, :name)
+    @order_param = params[:sort_by] || Constants::DEFAULT_BOOKS_ORDER
+
+    result = DefaultBooksQuery.call(category: params[:category], sort_param: params[:sort_by]).page(params[:page]).per(Constants::BOOKS_PER_PAGE)
```

Comment:
we can put params to the query and delegate sort_by, category and etc inside class

Reply:
check if **draper** can decorate the collection in another way